server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalTopHits.java

```diff
@@ -184,9 +185,42 @@ protected boolean mustReduceOnSingleInternalAgg() {
     public Object getProperty(List<String> path) {
         if (path.isEmpty()) {
             return this;
+        }
+        if (path.size() != 1) {
+            throw new IllegalArgumentException(
+                "property paths for top_hits ["
+                    + getName()
+                    + "] can only contain a single field in _source, score or sort values, got "
+                    + path
+            );
+        }
+
+        // Supported property prefixes.
+        final String SOURCE = "_source";
```

Comment:
maybe turn these into constants? (by turning these fields into static final fields on this class)